cl/tests/utils.py

```diff
@@ -1,20 +1,155 @@
 from typing import Tuple
 
+from django.conf import settings
 from django.contrib.auth.models import User
 from django.core.files.base import ContentFile
+from django.http import HttpHeaders  # type: ignore[attr-defined]
+from django.test import AsyncClient
+from django.utils.encoding import force_bytes
+from django.utils.http import urlencode
 from requests import Response
 from rest_framework.authtoken.models import Token
-from rest_framework.test import APIClient
+from rest_framework.settings import api_settings
 from selenium.webdriver.remote.webelement import WebElement
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-def make_client(user_pk: int) -> APIClient:
+class AsyncAPIClient(AsyncClient):
```

Comment:
It looks like you used pieces from `APIRequestFactory` + `APIClient` to make this. Why not override those instead of duplicating their code? If you did that, couldn't you avoid repeating the `__init__`, `_eoncode_data`, and `credentials` methods?

Reply:
I had to rework credentials to make it work with AsyncClient and `_encode_data` was a private API so seemed best to copy it instead.

Reply:
The other nice thing about using those is that it'll help us track down where the code is borrowing from in the future. As it is, I had to carefully compare your `_encode_data` method vs. the source to see what you were up to (turns out nothing, right?). Much better not to repeat it, and use inheritance, I think.

Reply:
> Much better not to repeat it, and use inheritance, I think.

I think I avoided using inheritance since `APIClient` wasn't really designed for async usage which made things tricky.

Reply:
I tried doing some overrides with async methods and it seemed to work fine? 

![image](https://github.com/freelawproject/courtlistener/assets/236970/f881a658-7214-461d-81e7-d33c6868e8c7)

Reply:
So if `Foo` has private method `_encode_data` called by `Foo.get()` then I don't think you are supposed to call `_encode_data` from `Bar.get()` if it's private to `Foo` right?

Reply:
That feels right from a Java-esque perspective of private methods, but I always think of Django's private methods as "Use it, but it might disappear or change someday." I think that's fine in this case, since tests would catch that when we upgraded.